packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -760,27 +762,43 @@ export class Function extends FunctionBase {
     return this._logGroup;
   }
 
+  public checkEdgeCompatibility(): void {
+    // Check env vars
+    const envEntries = Object.entries(this.environment);
+    for (const [key, config] of envEntries) {
+      if (config.removeInEdge) {
+        delete this.environment[key];
+        this.node.addInfo(`Removed ${key} environment variable for Lambda@Edge compatibility`);
+      }
+    }
+    const envKeys = Object.keys(this.environment);
+    if (envKeys.length !== 0) {
+      throw new Error(`The function ${this.node.path} is not compatible for Lambda@Edge because it contains environment variables that are not marked for removal when used for Lambda@Edge: ${envKeys}. Use \`addEnvironment()\` to allow edge removal.`);
+    }
+
+    // TODO: additional checks, see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-requirements-limits.html#lambda-requirements-lambda-function-configuration
+
```

Comment:
Prefer not to keep todos in code. They're not tracked and hence not useful.

Open an issue once this is merged so we can feed this into our planning, or have other folks discover and implement.